packages/starknet-snap/src/executeTxn.ts

```diff
@@ -1,28 +1,73 @@
 import { toJson } from './utils/serializer';
-import { getNetworkFromChainId, getTxnSnapTxt } from './utils/snapUtils';
-import { getKeysFromAddress, executeTxn as executeTxnUtil } from './utils/starknetUtils';
-import { ApiParams, ExecuteTxnRequestParams } from './types/snapApi';
+import { CallDetails } from 'starknet';
+import { getNetworkFromChainId, getTxnSnapTxt, addDialogTxt } from './utils/snapUtils';
+import { getKeysFromAddress, executeTxn as executeTxnUtil, isAccountDeployed } from './utils/starknetUtils';
+import { ApiParams, EstimateFeeRequestParams, ExecuteTxnRequestParams } from './types/snapApi';
+import { createAccount } from './createAccount';
 import { DialogType } from '@metamask/rpc-methods';
-import { heading, panel } from '@metamask/snaps-sdk';
+import { heading, panel, divider } from '@metamask/snaps-sdk';
 import { logger } from './utils/logger';
+import { estimateFee } from './estimateFee';
 
 export async function executeTxn(params: ApiParams) {
   try {
     const { state, keyDeriver, requestParams, wallet } = params;
     const requestParamsObj = requestParams as ExecuteTxnRequestParams;
+    const requestParamsObjEstimateFee = requestParams as EstimateFeeRequestParams;
 
-    logger.log(`executeTxn params: ${toJson(requestParamsObj, 2)}}`);
+    const txnInvocation = (
+      (requestParamsObj.txnInvocation as CallDetails).calldata
+        ? requestParamsObj.txnInvocation
+        : requestParamsObj.txnInvocation[0]
+    ) as CallDetails;
+
+    requestParamsObjEstimateFee.senderAddress = requestParamsObj.senderAddress;
+    requestParamsObjEstimateFee.contractAddress = txnInvocation.contractAddress;
+    requestParamsObjEstimateFee.contractFuncName = txnInvocation.entrypoint;
+    requestParamsObjEstimateFee.contractCallData = (txnInvocation.calldata as string[]).join(',');
+
+    logger.log(`executeTxn params: ${toJson(requestParamsObj, 2)}`);
+    const estimateFeeResp = await estimateFee({ ...params, ...requestParamsObjEstimateFee });
```

Comment:
in that case we may use estimateFees 

and build the Invocations in this file

1. init a Invocations base on the request
1. check if account deployed
2. if not , then append left in Invocations for create account
3. run estimateFeeBulk , i think we really dont need to call the estimateFees
4. run addFeesFromAllTransactions(estimateBulkFeeResp); , which can sum the fee for you
5. if yes, just run estimateFeeUtil, again i think we really dont need to call the estimateFees
6. with the fee, we add the gas fee in the snapComponents
7. and display the dialog
8. and process the txn

Reply:
I adapted the code to run estimateFeeBulk directly from executeTxn.